bindings/rascal/utils/cur.py

```diff
@@ -67,7 +69,9 @@ def __init__(self, representation, Nselect, act_on='sample per species', is_dete
         if act_on in ['sample', 'sample per species', 'feature']:
             self.act_on = act_on
         else:
-            raise 'Wrong input: {}'.format(act_on)
+            raise NotImplementedError(
```

Comment:
`__init__` should never raise a `NotImplementedError` --  this looks like more of a `ValueError` to me.

Reply:
> __init__ should never raise a NotImplementedError

why not ?

Reply:
Because it would suggest that `__init__` isn't implemented, which is nonsense. In fact, [`NotImplementedError`](https://docs.python.org/3/library/exceptions.html#NotImplementedError) is only intended for the very specific purpose of indicating "abstract" methods in base classes. (The similarly-named [`NotImplemented`](https://docs.python.org/3/library/constants.html#NotImplemented) has a different, also very specific purpose.)

The correct exception for indicating a problem with the input parameters of a method is [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError).